test/standalone/c_compiler/build.zig

```diff
@@ -23,15 +20,30 @@ pub fn build(b: *Builder) void {
     exe_c.setBuildMode(mode);
     exe_c.setTarget(target);
     exe_c.linkLibC();
+    exe_c.enable_wine = is_wine_enabled;
+    exe_c.enable_qemu = is_qemu_enabled;
+    exe_c.enable_wasmtime = is_wasmtime_enabled;
+    exe_c.enable_darling = is_darling_enabled;
 
     const exe_cpp = b.addExecutable("test_cpp", null);
     b.default_step.dependOn(&exe_cpp.step);
     exe_cpp.addCSourceFile("test.cpp", &[0][]const u8{});
     exe_cpp.setBuildMode(mode);
     exe_cpp.setTarget(target);
-    exe_cpp.linkSystemLibrary("c++");
+    exe_cpp.linkLibCpp();
+    exe_cpp.single_threaded = single_threaded;
+    const os_tag = target.getOsTag();
+    // macos C++ exceptions could be compiled, but not being catched,
+    // additional support is required, possibly unwind + DWARF CFI
+    if (target.getCpuArch().isWasm() or os_tag == .macos) {
+        exe_cpp.defineCMacro("_LIBCPP_NO_EXCEPTIONS", null);
+    }
```

Comment:
What's going on here? this define logic belongs in the compiler, not in this build.zig file.

Reply:
As you know, C++ standard library (use older STL acronym later) heavily uses exceptions, and I've updated the test to use them (as it could be a key functionality for C++ applications like in my use-case). Unfortunately, wasm and macOS (currently) has one issue that I'm not quite sure how to overcome: the code is compiled, but doesn't work correctly. I guess the proper solution would be to stop compiling the code if compiler detects that exceptions are used for wasm and macOS, but it means that C++ STL won't be compiled at all on those platforms, and that would be quite breaking change (as STL still could be used with incorrectly working exceptions if used carefully). I didn't want to risk introducing it as it could affect way too many users.

So, for now I've just declared a macro (it could have any name though) that disables exception code in the test. I was going to fix macOS exception handling later if you team won't have time for that, but for now I've just inserted that stub.